src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3229 @@
+        /// </summary>
+        private static readonly Tuple<string, Version> s_defaultModuleInfoForResource = new Tuple<string, Version>("PSDesiredStateConfiguration", new Version(1, 1));
+
+        /// <summary>
+        /// Default ModuleName and ModuleVersion to use for meta configuration resources.
+        /// </summary>
+        internal static readonly Tuple<string, Version> DefaultModuleInfoForMetaConfigResource = new Tuple<string, Version>("PSDesiredStateConfigurationEngine", new Version(2, 0));
+
+        /// <summary>
+        /// A set of dynamic keywords that can be used in both configuration and meta configuration.
+        /// </summary>
+        internal static readonly HashSet<string> SystemResourceNames =
+            new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "Node", "OMI_ConfigurationDocument" };
+
+        /// <summary>
+        /// When this property is set to true, DSC Cache will cache multiple versions of a resource.
+        /// That means it will cache duplicate resource classes (class names for a resource in two different module versions are same).
+        /// NOTE: This property should be set to false for DSC compiler related methods/functionality, such as Import-DscResource,
+        ///       because the Mof serializer does not support deserialization of classes with different versions.
+        /// </summary>
+        [ThreadStatic]
+        private static bool t_cacheResourcesFromMultipleModuleVersions;
+
+        private static bool CacheResourcesFromMultipleModuleVersions
+        {
+            get
+            {
+                return t_cacheResourcesFromMultipleModuleVersions;
+            }
+
+            set
+            {
+                t_cacheResourcesFromMultipleModuleVersions = value;
+            }
+        }
+
+        /// <summary>
+        /// Initialize the class cache with the default classes in $ENV:SystemDirectory\Configuration.
+        /// </summary>
+        public static void Initialize()
+        {
+            Initialize(errors: null, modulePathList: null);
+        }
+
+        /// <summary>
+        /// Initialize the class cache with the default classes in $ENV:SystemDirectory\Configuration.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">List of module path from where DSC PS modules will be loaded.</param>
+        public static void Initialize(Collection<Exception> errors, List<string> modulePathList)
+        {
+            s_tracer.WriteLine("Initializing DSC class cache");
+
+            if (Platform.IsLinux || Platform.IsMacOS)
+            {
+                // Load the base schema files.
+                ClearCache();
+                var dscConfigurationDirectory = Environment.GetEnvironmentVariable("DSC_HOME") ??
+                                                "/etc/opt/omi/conf/dsc/configuration";
```

Comment:
We should decide if this path still makes sense if we are no longer dependent on OMI

Reply:
This is essentially a DSC design question. Based on what I've heard from people who have more experience with DSC, this path no longer makes sense. [We discussed this with Travis](https://github.com/PowerShell/PowerShell/pull/13399#discussion_r468837553).
Updated.